src/cryptonote_core/tx_pool.cpp

```diff
@@ -375,16 +375,26 @@ namespace cryptonote
         txpool_tx_meta_t meta;
         if (!m_blockchain.get_txpool_tx_meta(txid, meta))
         {
-          MERROR("Failed to find tx in txpool");
+          MERROR("Failed to find tx_meta in txpool");
           return;
         }
         // don't prune the kept_by_block ones, they're likely added because we're adding a block with those
-        if (meta.kept_by_block)
+        switch (meta.get_relay_method())
         {
-          --it;
-          continue;
+          default:
+            MERROR("Failed to find tx in txpool");
+            return;
```

Comment:
Leftover from the merging of error code and method ?

Reply:
That error message is cruft but the return was intended. I suppose in this context isn't irrelevant - I'll switch back to an `if`.